pkg/k8s/service_cache.go

```diff
@@ -133,15 +134,15 @@ func (s *ServiceCache) GetServiceIP(svcID ServiceID) *loadbalancer.L3n4Addr {
 }
 
 // GetServiceFrontendIP returns the frontend IP (aka clusterIP) for the given service with type.
-func (s *ServiceCache) GetServiceFrontendIP(svcID ServiceID, svcType loadbalancer.SVCType) net.IP {
+func (s *ServiceCache) GetServiceFrontendIP(svcID ServiceID, svcType loadbalancer.SVCType) (net.IP, error) {
 	s.mutex.RLock()
 	defer s.mutex.RUnlock()
 	svc := s.services[svcID]
 	if svc == nil || svc.Type != svcType || len(svc.FrontendIPs) == 0 {
-		return nil
+		return nil, fmt.Errorf("service %v not found", svcID)
```

Comment:
The caller already knows the svcID, so does it need to be formatted within the error msg?